core/src/main/scala/kafka/controller/KafkaController.scala

```diff
@@ -2665,6 +2693,9 @@ class KafkaController(val config: KafkaConfig,
         partitionsBeingDeleted.map(
           _ -> Left(new ApiError(Errors.INVALID_TOPIC_EXCEPTION, "The topic is being deleted"))
         ) ++
+        knownPartitionsWithInvalidRequest.map(
+          _ -> Left(new ApiError(Errors.INVALID_REQUEST, "No recommended leader provided for the partition"))
+        ) ++
```

Comment:
I don't have enough context to understand how/where this is used, but it seems to confirm my previous (confused) comment that this recommended-leader request, which I'd expect to be advisory/best-effort only (with transparent fallback to the existing flavor of leadership election for any partitions that need it), is somehow being treated as an absolute requirement-request that triggers errors if less than 100% of the partitions have recommended leaders or fall short of perfect knowledge in any other way. Is this stuff being logged, and if so, at warning or error level, or only at info?  Anything higher than info seems like a problem to me, but I'm not following the complete flow, so maybe I'm worrying about something that's not important.

Reply:
As discussed offline, there would be a top level error code, and partition level error code in the ElectLeadersResponse.
The ElectLeadersRequest is not transactional, so some partitions may succeed and some may fail.
The knownPartitionsWithInvalidRequest are logged at warning level.